bin/ufmt.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env nash
+
+# universal formatter
+
+import "io"
+import "acme/acme.sh"
+
+fn getext(path) {
+	var base, status <= basename $path
+
+	abortonerr($status, "basename failed")
+
+	var parts <= split($base, ".")
+
+	if len($parts) == "1" {
+		return ""
+	}
+
+	var partsz <= len($parts)
+
+	var last, _ <= expr $partsz "-" 1
+
+	return $parts[$last]
+}
+
+fn main() {
+	var fname <= acme_fname()
+
+	var ext <= getext($fname)
+
+	if $ext == "sh" {
```

Comment:
Another idea to not depend on file extension is to use nashfmt and gofmt itself to detect the file type, since they fail if it is not valid code. But it can be too much overhead (on vim I used nashfmt and it usually runs pretty fast). The problem is that somethimes bash code also looks like nash code...so it did not completely solved the problem =/

Reply:
oh yeah. I dont like this approach, but we can implement to see how it goes.

In case of Go, I don't think makes sense use it with other extension.

But if the file has no extension, what do you think of look in the first line of source and see if it has a shebang? #!/bin/bash, #!/usr/bin/env nash, #!/usr/bin/env python, and so on.

Reply:
Looking the first line for scripted languages seems a good compromise =)